Makefile

```diff
@@ -31,9 +31,20 @@ REPO_URL ?= https://github.com/packethost/cluster-api-provider-packet
 BUILDARCH ?= $(shell uname -m)
 BUILDOS ?= $(shell uname -s | tr A-Z a-z)
 
-TEST_E2E_DIR := test/e2e
 E2E_FOCUS := "functional tests"
 
+# Directories.
+TOOLS_DIR := hack/tools
+TOOLS_BIN_DIR := $(TOOLS_DIR)/bin
+BIN_DIR := bin
+TEST_E2E_DIR := test/e2e
+
+# Binaries.
+KUSTOMIZE := $(TOOLS_BIN_DIR)/kustomize
```

Comment:
Can we have a "human-readable" dependency, so it looks something like:

```makefile
kustomize: $(KUSTOMIZE)
$(KUSTOMIZE): $(TOOLS_DIR)/go.mod # Build kustomize from tools folder.
 	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/kustomize sigs.k8s.io/kustomize/kustomize/v3
```

and then for the later dependency:

```makefile
manifest: kustomize semver release-manifests release-clusterctl release-cluster-template
```

and then we can have a list of tools:

```makefile
tools: kustomize controller-gen kubebuilder
```

This makes it easier to read, easier to parse, and one can do:

```
make kustomize
```

 to ensure it is installed.